File: Java/10_Ud2_IntentExplicito1_ClasesJava_v2/MainActivity.java
package com.pms.a10_ud2intentexplicito01;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    // objetos View del Layout
    EditText txtNombre;
    Button btnHola;
    //realmente, el totón, no es necesario declararlo, pues el evento onClick se controla desde el
    //método declarado en la propiedad del layout

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // referencias a objetos del layout
        txtNombre = (EditText) findViewById(R.id.TxtNombre);
        //btnHola = (Button) findViewById(R.id.BtnHola);
    }

    /**
     * Funcionalidad o comportamiento al pulsar el botón BtnHola
     * @param v: botón btnHola
     */
    public void bHola (View v) {

        //1) intent para lanzar la activity SegundaActivity
        Intent inte2 = new Intent(getApplicationContext(),
                SegundaActivity.class);


        //2) Se añade información EXTRA mediante a) un objeto Bundle o  b) el propio Intent

        //a) PASANDO información EXTRA con un OBJETO Bundle (colección de parejas  clave-valor)

        //Objeto bundle para pasar información a la activity que se lanza
        Bundle bundatos = new Bundle();
        // información pasada:
        // clave:"NOMBRE",
        // valor=txtNombre.getText().toString()
        bundatos.putString("NOMBRE", txtNombre.getText().toString());
        // asocia información pasada al intent que lanzará la nueva activity
        inte2.putExtras(bundatos);

   /*    //b) PASANDO información EXTRA mediante el propio Intent (parejas clave-valoe)

        // información pasada:
        // clave:"NOMBRE",
        // valor=txtNombre.getText().toString()
       inte2.putExtra("NOMBRE", txtNombre.getText().toString());

*/
        //3) se lanza la activity asociada a este Intent con los Extra añadidos, si los hay
        startActivity(inte2);



    }
}

File: Kotlin/10_Ud2_IntentExplicito1_ClasesJava_v2/MainActivity.kt
package com.pms.a10_ud2intentexplicito01

class MainActivity : AppCompatActivity() {
    // objetos View del Layout
    var txtNombre: EditText? = null
    var btnHola: android.widget.Button? = null

    //realmente, el totón, no es necesario declararlo, pues el evento onClick se controla desde el
    //método declarado en la propiedad del layout
    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // referencias a objetos del layout
        txtNombre = findViewById<android.view.View>(R.id.TxtNombre) as EditText?
        //btnHola = (Button) findViewById(R.id.BtnHola);
    }

    /**
     * Funcionalidad o comportamiento al pulsar el botón BtnHola
     * @param v: botón btnHola
     */
    fun bHola(v: android.view.View?) {

        //1) intent para lanzar la activity SegundaActivity
        val inte2 = Intent(getApplicationContext(),
                SegundaActivity::class.java)


        //2) Se añade información EXTRA mediante a) un objeto Bundle o  b) el propio Intent

        //a) PASANDO información EXTRA con un OBJETO Bundle (colección de parejas  clave-valor)

        //Objeto bundle para pasar información a la activity que se lanza
        val bundatos = Bundle()
        // información pasada:
        // clave:"NOMBRE",
        // valor=txtNombre.getText().toString()
        bundatos.putString("NOMBRE", txtNombre.getText().toString())
        // asocia información pasada al intent que lanzará la nueva activity
        inte2.putExtras(bundatos)

        /*    //b) PASANDO información EXTRA mediante el propio Intent (parejas clave-valoe)

        // información pasada:
        // clave:"NOMBRE",
        // valor=txtNombre.getText().toString()
       inte2.putExtra("NOMBRE", txtNombre.getText().toString());

*/
        //3) se lanza la activity asociada a este Intent con los Extra añadidos, si los hay
        startActivity(inte2)
    }
}